lib/index.js

```diff
@@ -2,22 +2,24 @@
 
 exports.register = (server, options, next) => {
 
+  options.defaultEnabled = options.defaultEnabled || false;
+
   server.ext('onPostAuth', (request, reply) => {
     request.plugins['hapi-rate-limit'] = { rate: null };
 
     if (!(request.route.method === 'get' || request.route.method === 'post' || request.route.method === 'delete')) {
       return reply.continue();
     }
 
-    let rate = Object.assign({}, options.defaultRate(request));
+    let rate;
     const routeSettings = request.route.settings.plugins.rateLimit;
 
-    if (routeSettings) {
-      if (routeSettings.enabled === false) {
-        return reply.continue();
-      }
-
-      rate = routeSettings.rate(request);
+    if (!routeSettings && options.defaultEnabled) {
+      rate = options.defaultRate(request);
+    } else if (routeSettings && routeSettings.enabled !== false) {
+      rate = routeSettings.rate ? routeSettings.rate(request) : options.defaultRate(request);
```

Comment:
I'd rather not introduce a known bug, we can add a check to see if `options.defaultEnabled` is set to avoid this issue.

```javascript 
...
} else if (routeSettings && routeSettings.enabled && options.defaultEnabled && options.defaultRate) { 
  ...
}
...
```